### PR TITLE
Add Traditional Chinese translation of ml&pl blog post

### PR DESCRIPTION
**Pull request by chengchingwen**
As he [commented](https://github.com/JuliaLang/julialang.github.com/pull/689#issuecomment-352428033) in #689 

@iblis17

### CODE REVIEW

File: blog/_posts/2017-12-25-ml&pl-zh_tw.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+## 那量身訂製的 ML 語言長什麼樣子？
+
+除了 ML 外，還有其他領域也碰到需要語言層級的設計問題，但這並非是史無前例的。在像是[形式化推論及驗證](https://coq.inria.fr/)或是[叢集運算](https://chapel-lang.org/)的領域上，新量身訂做的語言已經證實了是有效的解決方案。同樣的，我們期望看到一個新的或是現存的語言對 ML 的數值系統，可微分，可平行且甚至是機率計算的需求做客製化。
+
+對於 ML 語言，一個明顯的挑戰是同時達到通用性以及高效能，而提早混合的方式將會需要更大量的開發工作。我們希望未來的 ML runtimes 將會需要支援隨心所欲的混合各種方法(靜態的計算圖嵌入動態計算圖又嵌入靜態計算圖…）以及在部屬環境讓編譯動態程式碼做的更好。理想上，只會剩下單一種彈性的圖格式（或是 AST)。這個 AST 需要有能夠靜態描述動態行為的語法（比如：一個 `for` loop) ― 換句話說，他應該看起來更像是個標準的程式語言。
+
+*可程式化語意*將會開啟彈性的新階段， 而且可以提供一個類似 macros 的功能。 這將會允許類似多 GPU　訓練功能可以被建立在核心系統上，並且指定純資料流語意的程式碼（相對的是標準命令式語意，雖然較為彈性但是容易有負作用，對於優化來說較為不安全)。他同樣可以用來操作 probailistic programming ，或是自然語言處理模型常手動實作的[向量化](https://www.cs.cmu.edu/~guyb/papers/Ble90.pdf)(批量處理）傳遞。
+
+同樣的在 PL 社群裡， ML 工程師應該要花更多時間注意傳統的自動微分（AD）社群。 ML 語言可以從頂尖的程式語言設計上得到[支援微分為真的第一公民（first-class）](https://arxiv.org/pdf/1611.03416.pdf)的靈感。這樣的語言可以簡單地混合符號和 runtime 技術（這可以幫忙以上提到的取捨）、混合前向及反向模式的 AD （以增進效能及記憶體使用），以及[區別不同的 GPU 核心](http://mikeinnes.github.io/2017/08/24/cudanative.html) ― 以上都沒有任何效能流失。
```
Comment (iblis17) [inline blog/_posts/2017-12-25-ml&pl-zh_tw.md]: ‘GPU 核心’ 這裡是 kernel?

**Comment by ViralBShah** [comment]
Should we publish this as a separate post or somehow make it a translation of the earlier post? It would seem that for someone coming for a Chinese language version, we would want to offer two options right there. Perhaps maybe as options on the English language version too.

File: blog/_posts/2017-12-25-ml&pl-zh_tw.md (hunk 1 above)
Comment (chengchingwen) [reply blog/_posts/2017-12-25-ml&pl-zh_tw.md]: 對 原文是 'GPU kernel'

Comment (iblis17) [reply blog/_posts/2017-12-25-ml&pl-zh_tw.md]: 那麼就不翻？

**Comment by iblis17** [comment]
> Should we publish this as a separate post or somehow make it a translation of the earlier post?

I'm fine with both, and I personally want this page owns an unique url, so we can spread it easily (but not sure how to achieving this gaol with Jekyll)

File: blog/_posts/2017-12-25-ml&pl-zh_tw.md (hunk 1 above)
Comment (chengchingwen) [reply blog/_posts/2017-12-25-ml&pl-zh_tw.md]: 改了

**Comment by ViralBShah** [comment]
Ok, let's have it have its own page for now. Is it good to merge?

File: blog/_posts/2017-12-25-ml&pl-zh_tw.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+MXNet 的 [Gluon](https://mxnet.incubator.apache.org/api/python/gluon.html) 就是在努力找到一個方式對兩邊都最好，至少到一定程度上。這個想法是透過結合基本的動態自動微分及程式碼追蹤方式，這個方式會製造可優化的"靜態子圖"。不幸的是，這是某種完全不同的實做以及 API 的混合體。他也同樣有侷限性。MXNet 使用他的圖不只在核心層級的優化還有高階的圖排程，像是[將一個模型分散到數個 GPU 上](https://mxnet.incubator.apache.org/how_to/multi_devices.html)。這個混合體要如何處理這些並不清楚，除了加入其他新的 API 讓圖容器可以被運算節點動態計算。
+
+## 那量身訂製的 ML 語言長什麼樣子？
+
+除了 ML 外，還有其他領域也碰到需要語言層級的設計問題，但這並非是史無前例的。在像是[形式化推論及驗證](https://coq.inria.fr/)或是[叢集運算](https://chapel-lang.org/)的領域上，新量身訂做的語言已經證實了是有效的解決方案。同樣的，我們期望看到一個新的或是現存的語言對 ML 的數值系統，可微分，可平行且甚至是機率計算的需求做客製化。
+
+對於 ML 語言，一個明顯的挑戰是同時達到通用性以及高效能，而提早混合的方式將會需要更大量的開發工作。我們希望未來的 ML runtimes 將會需要支援隨心所欲的混合各種方法(靜態的計算圖嵌入動態計算圖又嵌入靜態計算圖…）以及在部屬環境讓編譯動態程式碼做的更好。理想上，只會剩下單一種彈性的圖格式（或是 AST)。這個 AST 需要有能夠靜態描述動態行為的語法（比如：一個 `for` loop) ― 換句話說，他應該看起來更像是個標準的程式語言。
```
Comment (iblis17) [inline blog/_posts/2017-12-25-ml&pl-zh_tw.md]: 圖格式好怪？

Comment (chengchingwen) [reply blog/_posts/2017-12-25-ml&pl-zh_tw.md]: 有什麼比較好的翻法嗎？原文是 "graph format"

Comment (iblis17) [reply blog/_posts/2017-12-25-ml&pl-zh_tw.md]: 有辦法保留整篇的 graph 嗎？
從上面的計算圖到下面的圖格式，我覺得都不翻譯

Comment (chengchingwen) [reply blog/_posts/2017-12-25-ml&pl-zh_tw.md]: 也是可以 但其他部份的 graph 的詞要一起不翻嗎？ sub-graph or computational graph 這種

Comment (iblis17) [reply blog/_posts/2017-12-25-ml&pl-zh_tw.md]: 對啊

Comment (chengchingwen) [reply blog/_posts/2017-12-25-ml&pl-zh_tw.md]: 這樣子呢？

**Pull request merged**